### PR TITLE
Avoid rdeps when using hash all targets

### PR DESCRIPTION
**Pull request by tinder-maxwellelliott**
When using hash all targets it is overkill to use rdeps to query the
impacted targets, since we know at the end of hashing what was
affected.

### CODE REVIEW

**Review by purkhusid — APPROVED** [review]
👍

File: src/main/java/com/bazel_diff/TargetHashingClient.java [hunk 1]
```diff
@@ -45,7 +46,7 @@
                 impactedTargets.add(entry.getKey());
             }
         }
-        return bazelClient.queryForImpactedTargets(impactedTargets, avoidQuery);
+        return bazelClient.queryForImpactedTargets(impactedTargets, avoidQuery, hashAllTargets);
```
Comment (TimothySeah) [inline src/main/java/com/bazel_diff/TargetHashingClient.java:49]: If I'm understanding things correctly, I think you can just return impactedTargets here if you hash all targets, but I haven't tested this out.

Comment (tinder-maxwellelliott) [reply src/main/java/com/bazel_diff/TargetHashingClient.java:49]: We could, yet the user could potentially provide `avoidQuery`, we should still respect that value

Comment (TimothySeah) [reply src/main/java/com/bazel_diff/TargetHashingClient.java:49]: In that case would it make sense to return "bazelClient.query..." if there's an avoidQuery and impactedTargets if there's no avoidQuery?

**Pull request merged**